simpledb/sstable_manager.go

```diff
@@ -117,28 +116,67 @@ func (s *SSTableManager) candidateTablesForCompaction(compactionMaxSizeBytes uin
 	s.managerLock.RLock()
 	defer s.managerLock.RUnlock()
 
+	var selectedForCompaction []bool
+	for i := 0; i < len(s.allSSTableReaders); i++ {
+		selectedForCompaction = append(selectedForCompaction, s.allSSTableReaders[i].MetaData().TotalBytes < compactionMaxSizeBytes)
+	}
+
+	/*
+		Imagine the following scenario we have with tombstones:
+			- Table 1 (size 2gig)
+			- Table 2 (size 1gig)
+			- Table 3 (size 3gig)
+			- Table 4 (size 1gig)
+
+		Given maxSize=2gig, the above selection would select table 2 and 4 as candidates.
+		Which will create an entirely wrong data lineage, as the changes from Table 3 will be completely ignored and an
+		entire new Table 2 will be created with "changes from the future".
+
+		The below function will fill the holes, by also selecting the tables in between for compaction - so the lineage is always preserved.
```

Comment:
even if you add the table 2,3 and 4 this selection prevent sending the tombstone to the first table. This make the physical deletion of the data in table 1 impossible. The database will grow indefinitely without any possibility to shrink. I think that size selection is incompatible with the possibility of shrinking. The only case when size selection is possible is when the sstable contains no tombstone.

Reply:
https://github.com/thomasjungblut/go-sstables/issues/36#issuecomment-2480501544

I think that this note is still actual. What about a separate Garbage collection process ?